### PR TITLE
Fix/refactor and test

### PR DESCRIPTION
**Pull request by Jacksonmcguire**
## What does this PR do?

- Begins filling out README, Writes and passes sidebar + form tests. Adds minor styling and refactor changes. Adds dynamic functionality for the category buttons

## How should it be tested in the UI?

- Fetching down the branch, running the tests, seeing if page is behaving as expected on a browser.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,5 +1,50 @@
...
+* To access the site on your local machine:
+  * Clone down the repo through the command line
+  * Run ```$npm install``` to install dependencies
+  * Run ```$npm start``` to open the page on a local browser
+* Otherwise the site is deployed and accessible [here]() 
+
+## Running Tests
+* After completing the setup steps above:
+  * Run ```$npx cypress open``` to open up the test suite
+  * From there click on the `metAPI-testing` file to open it in your browser and watch the tests run automatically
```
Comment (itsnameissteven) [inline README.md:50]: Thanks for doing this.

File: src/components/FilterForm/FilterForm.tsx [hunk 2]
```diff
@@ -61,67 +64,61 @@ export class FilterForm extends React.Component<FilterProps> {
...
           <option value='true'>HTTPS</option>
           <option value='false'>No HTTPS</option>
         </select>
-
-        <label htmlFor="Cors">Cors:</label>
+        <label htmlFor="Cors"></label>
         <select id="Cors" name="Cors" onChange={e => this.handleChange(e)}>
-          <option value='all'>--All--</option>
+          <option value='all'>Cors: --All--</option>
           <option value='yes'>Yes</option>
```
Comment (cbmackintosh) [inline src/components/FilterForm/FilterForm.tsx:118]: Good call here - keeping the labels without any innerText for accessibility. Moving the actual visual label into the dropdown. Saves space, looks a lot cleaner. Works better with screen responsiveness.

File: src/components/Category/Category.tsx [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+}
+
+export const Category = ({category, categoryLength, handleCatSelection}:categoryProps) => {
+  return (
+    <div key={category} className="category">
+      <input type="checkbox" id={category} name={category} onChange={e => handleCatSelection(e)} />
+      <label htmlFor={category}>{category + `(${categoryLength})`}</label>
+    </div>
+  )
+}
```
Comment (cbmackintosh) [inline src/components/Category/Category.tsx:17]: Good idea breaking this out into a separate component. Code's much more readable.

File: src/components/Category/Category.tsx [hunk 4]
```diff
@@ -0,0 +1,17 @@
+import React from 'react'
+import './Category.css'
+
+type categoryProps = {
+  category: string;
+  categoryLength: number;
+  handleCatSelection(e:React.ChangeEvent<HTMLInputElement>):void; 
+}
```
Comment (itsnameissteven) [inline src/components/Category/Category.tsx:8]: I haven't seen a prop declaration like this. This looks better than the arrow function.

**Review by cbmackintosh — APPROVED** [review]
Thanks for writing all those tests. Testing library looks pretty robust

**Review by itsnameissteven — APPROVED** [review]
Everything looks and works great. good to merge

**Pull request merged**